src/jit/compiler.h

```diff
@@ -4803,21 +4807,16 @@ class Compiler
     BlockSet_ValRet_T fgDomFindStartNodes(); // Computes which basic blocks don't have incoming edges in the flow graph.
                                              // Returns this as a set.
 
-    BlockSet_ValRet_T fgDomTreeEntryNodes(BasicBlockList** domTree); // Computes which nodes in the dominance forest are
-                                                                     // root nodes. Returns this as a set.
-
-#ifdef DEBUG
-    void fgDispDomTree(BasicBlockList** domTree); // Helper that prints out the Dominator Tree in debug builds.
-#endif                                            // DEBUG
+    INDEBUG(void fgDispDomTree(DomTreeNode* domTree);) // Helper that prints out the Dominator Tree in debug builds.
 
-    void fgBuildDomTree(); // Once we compute all the immediate dominator sets for each node in the flow graph
-                           // (performed by fgComputeDoms), this procedure builds the dominance tree represented
-                           // adjacency lists.
+    DomTreeNode* fgBuildDomTree(); // Once we compute all the immediate dominator sets for each node in the flow graph
```

Comment:
Oh, that comment, I think, is incorrect, because `fgComputeDoms` doesn't only compute `the immediate dominator sets` but also calls `fgComputeDoms`, so `fgComputeDoms` builds everything.

and `fgBuildDomTree` is also called not after (inside)  `fgComputeDoms` but also after `ComputeImmediateDom(postOrder, count);` in `SSABuilder::Build()`.

I agree that it is more a flowgraph issue than an SSA issue, but could you please fix the comments while you are changing these functions signatures?

Reply:
OK, I'll try to adjust comments a bit. BTW, `fgComputeDoms` comment also says something that's not quite right: "We first assume the fields bbIDom on each basic block are invalid". That's not true, it just sets all `bbIDom` to null or the imaginary root before doing anything else.